lib/api/gitlab.js

```diff
@@ -0,0 +1,296 @@
+const logger = require('winston');
+const glGot = require('gl-got');
+
+const config = {};
+
+module.exports = {
+  initRepo,
+  // Search
+  findFilePaths,
+  // Branch
+  branchExists,
+  getBranchPr,
+  // issue
+  addAssignees,
+  addReviewers,
+  addLabels,
+  // PR
+  findPr,
+  checkForClosedPr,
+  createPr,
+  getPr,
+  updatePr,
+  // file
+  commitFileToBranch,
+  getFile,
+  getFileContent,
+  getFileJson,
+};
+
+// Initialize GitLab by getting base branch
+async function initRepo(repoName, token) {
+  logger.debug(`initRepo(${repoName})`);
+  if (token) {
+    process.env.GITLAB_TOKEN = token;
+  } else if (!process.env.GITLAB_TOKEN) {
+    throw new Error(`No token found for GitHub repository ${repoName}`);
+  }
+  config.repoName = repoName.replace('/', '%2F');
+  try {
+    const res = await glGot(`projects/${config.repoName}`);
+    config.defaultBranch = res.body.default_branch;
+    logger.debug(`${repoName} default branch = ${config.defaultBranch}`);
+  } catch (err) {
+    logger.error(`GitHub init error: ${JSON.stringify(err)}`);
+    throw err;
+  }
+}
+
+// Search
+
+// Returns an array of file paths in current repo matching the fileName
+async function findFilePaths(fileName) {
+  logger.verbose('Can\'t find multiple package.json files in GitLab');
+  return [fileName];
+}
+
+// Branch
+
+// Returns true if branch exists, otherwise false
+async function branchExists(branchName) {
+  logger.debug(`Checking if branch exists: ${branchName}`);
+  try {
+    const url = `projects/${config.repoName}/repository/branches/${branchName}`;
```

Comment:
All uses of `gl-got` expect that the project is on `gitlab.com`.

It looks like I can set `GITLAB_ENDPOINT` environment variable.

Could that environment variable be exposed through a CLI argument, perhaps `--endpoint`, similar to how `token` is exposed as `--token` and then used to set the `TOKEN` environment variable?